aten/src/ATen/native/quantized/cpu/quant_utils.h

```diff
@@ -28,6 +28,9 @@ inline TensorQuantizationParams ChooseQuantizationParams(
     bool preserve_sparsity = false,
     bool force_scale_power_of_two = false,
     bool reduce_range = false) {
+  TORCH_CHECK(
+      min <= max,
+      "In ChooseQuantizationParams, min should be less than or equal to max");
```

Comment:
if `min == max`, wouldn't `scale` be 0?

Reply:
Check https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/quant_utils.h#L58-L60
```
  // If scale is 0 or too small so its reciprocal is infinity, we arbitrary
  // adjust the scale to 0.1 . We want to avoid scale's reciprocal being	  // adjust the scale to 0.1 . We want to avoid scale's reciprocal being
  // infinity because some of fbgemm code pre-computes scale's reciprocal to do	  // infinity because some of fbgemm code pre-computes scale's reciprocal to do
  // multiplication instead of division in the time critical part of code.	  // multiplication instead of division in the time critical part of code.
  if (scale == 0.0f || isinf(1.0f / scale)) {	  if (scale == 0.0f || isinf(1.0f / scale)) {
    scale = 0.1;	    scale = 0.1;
  }	  }
```

If `min == max`, we first get `scale = 0` and then adjust it to `0.1` before checking `scale > 0`.